### PR TITLE
chore(telegraf): add Avro input format (parser).

### LINKED ISSUE #5054: Telegraf parser docs aren't consistent with the available parsers
https://docs.influxdata.com/telegraf/v1.27/data_formats/input telegraf documentation for parsers does not match the actually available parsers, nor does it match https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_INPUT.md
Is there any way to update the former?

[Slack Message](https://influxdata.slack.com/archives/C0VAV6N92/p1690369123086279?thread_ts=1690369123.086279&cid=C0VAV6N92)

### PR DESCRIPTION
**Pull request by jstirnaman**
Closes #5054

### CODE REVIEW

File: content/telegraf/v1.27/data_formats/input/_index.md [hunk 1]
```diff
@@ -9,9 +9,11 @@ menu:
     parent: Data formats
 ---
 
-Telegraf contains many general purpose plugins that use a configurable parser for parsing input data into [metrics][].  This allows, for example, the
-`kafka_consumer` input plugin to process messages formatted in InfluxDB Line
-Protocol or in JSON format. Telegraf supports the following input data formats:
+Telegraf contains many general purpose plugins that use a configurable parser for parsing input data into [metrics][].
```
Comment (sanderson) [inline content/telegraf/v1.27/data_formats/input/_index.md:12]:
```suggestion
Telegraf contains many general purpose plugins that use a configurable parser for parsing input data into metrics.
```

File: content/telegraf/v1.27/data_formats/input/_index.md [hunk 2]
```diff
@@ -9,9 +9,11 @@ menu:
...
+This allows input plugins such as [`kafka_consumer` input plugin](/telegraf/v1.27/plugins/#input-kafka_consumer)
```
Comment (sanderson) [inline content/telegraf/v1.27/data_formats/input/_index.md:13]:
```suggestion
This allows input plugins such as the [`kafka_consumer` plugin](/telegraf/v1.27/plugins/#input-kafka_consumer)
```

File: content/telegraf/v1.27/data_formats/input/_index.md [hunk 3]
```diff
@@ -9,9 +9,11 @@ menu:
...
+to consume and process different data formats, such as InfluxDB Line
+Protocol or JSON.
```
Comment (sanderson) [inline content/telegraf/v1.27/data_formats/input/_index.md:15]:
```suggestion
to consume and process different data formats, such as InfluxDB line
protocol or JSON.
```

File: content/telegraf/v1.27/data_formats/input/avro.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
+---
+title: Avro input data format
+description: Use the `avro` input data format to parse metrics from a message serialized as Avro binary or JSON format.
+menu:
+  telegraf_1_27_ref:
+
```
Comment (sanderson) [inline content/telegraf/v1.27/data_formats/input/avro.md:6]:
```suggestion
```

**Review by powersj — APPROVED** [review]
Thank you!

File: content/telegraf/v1.27/data_formats/input/avro.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+The Avro input data format parses messages serialized as [Avro](https://avro.apache.org/) format and encoded as binary or JSON.
+
+## Wire format
+
+Avro messages should conform to [Wire Format](https://docs.confluent.io/platform/current/schema-registry/fundamentals/serdes-develop/index.html#wire-format) using the following byte-mapping:
+
+| Bytes | Area       | Description                                      |
+| ----- | ---------- | ------------------------------------------------ |
+| 0     | Magic Byte | Confluent serialization format version number; currently always `0` |
```
Comment (sanderson) [inline content/telegraf/v1.27/data_formats/input/avro.md:20]:
```suggestion
| 0     | Magic Byte | Confluent serialization format version number; currently always `0`. |
```

**Review by sanderson — APPROVED** [review]
Looks good. Just some minor suggestions.

**Pull request merged**